### DISCUSSION TITLE
How to fix issue on installation

### DISCUSSION THREAD

**Discussion by mictian** [swayos/swayos.github.io#15] [Q&A] [answered]
Hi,
First of all, this project seems AWESOME (unfortunately I cannot verify this as I am unable to install it :cry: )
In any case thank you very much for all the work and dedication.

I must clarify that my knowledge of Linux in general and Debian in particular are super basic.
I just download and installed Debian 12, and tried to run the installation script.
As a result I go the following error: https://pastebin.com/t0axQE6z (note that paste is the swayos_err file generated by the installation script itself).

Regards,
Mictian

P.S. Sorry if this is not the right place for this question. Maybe I should create an issue. If this is the case pls let me know and I will do that.

**Comment by milgra** [comment, marked as answer]
Hi, it seems that ninja package is missing from the package list, try sudo apt-get install ninja-build or sudo apt-get install ninja first.

**Reply by mictian** [reply]
Hi @milgra ,

First of all, thank you for your reply and sorry for my super late response.

Indeed that solve the installation problem :smile: (`sudo apt-get install ninja-build`).
However once it finished the installation and I restarted the system, when I login the system get hanged up and there is nothing else I can do :sob:

I think I should say that I am running Debian 12 netinstal without any window manager on top of VirtualBox 6.1.

Which kind of logs can I take a look at in order to understand what is happening? (I do understand that this may not be a problem with SwayOS, so feel free to not answer this question).

Just to be on the same page, on which Debian did you installed SwayOS? I think I might be using the wrong base system to install SwayOS.

Thank you in advance.

Regards,
Mictian

**Reply by milgra** [reply]
Hi, I've just updated the install scripts and tested it on Debian 12 in QEMU with the DVD image, it worked aell, can you please re-try installing it? I didn't try it with the netinst image though.

**Comment by mictian** [comment]
Hi,
I just tried with [this](https://cdimage.debian.org/images/unofficial/non-free/images-including-firmware/bookworm_di_alpha1+nonfree/amd64/iso-dvd/) image, in other words Bookworm non free + firmware iso, and still the same result :cry: after login debian hangs up.

Obviously I  am doing something wrong, as my knowledge about Linux is very limited.
Long story short, I will try to see how to debug this issue, but I do not think this is a problem with SwayOS.

Thank you.

Regards,
Mictian

**Comment by milgra** [comment]
I just realized you are trying it on VirtualBox, maybe the selected gpu driver is the problem, it must be OpenGL 2 capable, I use QXL in QEMU, I don't know the gpu driver situation with VirtualBox but some other driver worth trying.

**Reply by mictian** [reply]
Good point. I will try it and let you know.
BTW I was thinking that maybe you installed debian first with another desktop manager, and then installed SwayOS. Is that the case?

**Comment by mictian** [comment]
Hi @milgra ,

I just wanted to share my findings so far.

* Regarding OpenGL, I installled Debian 12 with Gnome and then installed the `mesa-utils` package using the same VirtualBox configuration that I am using for SwayOS. With that I was able to get this information:

```
mictian@debian:~$ glxinfo | grep render
direct rendering: Yes
    GLX_MESA_copy_sub_buffer, GLX_MESA_query_renderer, GLX_MESA_swap_control, 
    GLX_MESA_query_renderer, GLX_OML_swap_method, GLX_SGIS_multisample, 
Extended renderer info (GLX_MESA_query_renderer):
OpenGL renderer string: llvmpipe (LLVM 15.0.6, 256 bits)
    GL_ARB_conditional_render_inverted, GL_ARB_conservative_depth, 
    GL_NV_conditional_render, GL_NV_copy_image, GL_NV_depth_clamp, 
    GL_ARB_conditional_render_inverted, GL_ARB_conservative_depth, 
    GL_NV_ES1_1_compatibility, GL_NV_blend_square, GL_NV_conditional_render, 
    GL_EXT_read_format_bgra, GL_EXT_render_snorm, GL_EXT_robustness, 
    GL_NV_conditional_render, GL_NV_draw_buffers, GL_NV_fbo_color_attachments, 
    GL_OES_element_index_uint, GL_OES_fbo_render_mipmap, 
```

I interpret this as "OpenGL is working fine", please correct me if I am wrong.

* Regarding the fact that it hangs up once I login, I realize the problem is when `sway` tries to start. This shouldn't be a surprise as not many things happens at start up (but as I said before, I'm quite a newby so for me it was :yum: )
In other words [this line](https://github.com/swayos/swayos.github.io/blob/main/home/.zshrc#L41).

So I tried to run it this way `sway dbus-run-session sway --verbose` and I got this error:

![sway_error01](https://user-images.githubusercontent.com/6376906/211188885-4bdfb922-a1c8-4e84-aef2-01c7a855c321.png)

and I also tried run it like `sway --verbose` and I got this second error:

![sway_error02](https://user-images.githubusercontent.com/6376906/211188937-8fed5965-2c15-48a0-a510-5efd99597f23.png)

By any chance, do any of these issues ring a bell to you?

Regards,
Mictian

**Comment by milgra** [comment]
Try to run it with

`XDG_CURRENT_DESKTOP=sway dbus-run-session sway 
`
or just
`
dbus-run-session sway 
`
The should fix the first error.

The second error is strange, it shouldn't connect to seatd on Debian but you can try installing it.

**Reply by mictian** [reply]
As silly as it may sound, after installing `seatd` it just worked. Thank you very much :smile:

**Reply by milgra** [reply]
cool, no problem! :)
